### PR TITLE
fix_pytorch_exporter_validae_layer

### PR DESCRIPTION
**Pull request by eladc-git**

### CODE REVIEW

File: model_compression_toolkit/exporter/model_wrapper/pytorch/validate_layer.py [hunk 1]
```diff
@@ -33,10 +35,10 @@ def is_pytorch_layer_exportable(layer: Any) -> bool:
         """
         if isinstance(layer, PytorchQuantizationWrapper):
             quantizers = list(layer.weights_quantizers.values())
-            quantizers.extend(layer.activation_quantizers)
-            if all([isinstance(q, BasePyTorchInferableQuantizer) for q in quantizers]):
-                return True
-        return False
+            return all([isinstance(q, BasePyTorchInferableQuantizer) for q in quantizers])
+        elif isinstance(layer, PytorchActivationQuantizationHolder):
+            return isinstance(layer.activation_holder_quantizer, BasePyTorchInferableQuantizer)
```
Comment (reuvenperetz) [inline model_compression_toolkit/exporter/model_wrapper/pytorch/validate_layer.py:40]: Add check that every layer must be an nn.module.
Maybe it's better to raise an error with an informative error message in cases it's not a valid layer?

**Review by reuvenperetz — CHANGES_REQUESTED** [review]

File: model_compression_toolkit/exporter/model_wrapper/pytorch/validate_layer.py [hunk 2]
```diff
@@ -31,12 +34,35 @@ def is_pytorch_layer_exportable(layer: Any) -> bool:
...
         if isinstance(layer, PytorchQuantizationWrapper):
-            quantizers = list(layer.weights_quantizers.values())
-            quantizers.extend(layer.activation_quantizers)
-            if all([isinstance(q, BasePyTorchInferableQuantizer) for q in quantizers]):
-                return True
-        return False
+            valid_weights_quantizers = isinstance(layer.weights_quantizers, dict)
+            if not valid_weights_quantizers:
+                Logger.error(
+                    f'KerasQuantizationWrapper must have a weights_quantizers but has a '
```
Comment (reuvenperetz) [inline model_compression_toolkit/exporter/model_wrapper/pytorch/validate_layer.py:44]: It's not a KerasQuantizationWrapper

File: model_compression_toolkit/exporter/model_wrapper/pytorch/validate_layer.py [hunk 3]
```diff
@@ -31,12 +34,35 @@ def is_pytorch_layer_exportable(layer: Any) -> bool:
...
+                    f'{type(layer.weights_quantizers)} object') # pragma: no cover
+
+            if len(layer.weights_quantizers) == 0:
+                Logger.error(f'KerasQuantizationWrapper must have at least one weight quantizer, but found {len(layer.weights_quantizers)} quantizers.'
```
Comment (reuvenperetz) [inline model_compression_toolkit/exporter/model_wrapper/pytorch/validate_layer.py:48]: It's not a KerasQuantizationWrapper

File: model_compression_toolkit/exporter/model_wrapper/pytorch/validate_layer.py [hunk 4]
```diff
@@ -31,12 +34,35 @@ def is_pytorch_layer_exportable(layer: Any) -> bool:
         Returns:
             Check whether a PyTorch layer is a valid exportable layer or not.
         """
+        if not isinstance(layer, nn.Module):
+            Logger.error(f'Exportable layer must be a Pytorch layer, but layer {layer.name} is of type {type(layer)}') # pragma: no cover
```
Comment (reuvenperetz) [inline model_compression_toolkit/exporter/model_wrapper/pytorch/validate_layer.py:38]: Exportable layer must be a Pytorch *layer* -> Exportable layer must be a Pytorch *nn.Module*

**Review by reuvenperetz — APPROVED** [review]

**Review by reuvenperetz — APPROVED** [review]

File: model_compression_toolkit/exporter/model_wrapper/pytorch/validate_layer.py (hunk 3 above)
Comment (eladc-git) [reply model_compression_toolkit/exporter/model_wrapper/pytorch/validate_layer.py:48]: opsssss

**Pull request merged**